server/api/controllers/auth.py

```
@@ -255,3 +276,153 @@ def login():
+
+    if error:
+        return redirect(f"http://localhost:3000/profile?error={error}")
+
+    if not code or not state:
+        return redirect("http://localhost:3000/profile?error=missing_params")
+
+    # Verify state
+    try:
+        decoded_state = jwt.decode(state, JWT_SECRET, algorithms=["HS256"])
+        user_id = decoded_state["user_id"]
+    except:
```

Ruff (0.14.8)

Do not use bare `except`

(E722)